### ISSUE TITLE
support array field in BREAD system

### ISSUE THREAD

**Issue by ripples-alive** [thedevdojo/voyager#1695]
- Laravel Version: 5.4.30
- Voyager Version: 0.11.14
- PHP Version: 7.1.19
- Database Driver & Version: mysql 5.7

### Description:
I used the [attribute casting](https://laravel.com/docs/5.4/eloquent-mutators#attribute-casting) to automatically convert an `information` field to array.
When I try to use BREAD to read or edit the field (as `text_area`), an exception `htmlspecialchars() expects parameter 1 to be string, array given` is thrown.
Now I override the template `bread/read.blade.php` and `formfields/text_area.blade.php` to bypass.
But I'd like to know whether there is a better way or could be added as an feature later.

**Comment by lancepioch** [comment]
Labels: question, enhancement

**Comment by fletch3555** [comment]
Apologies for the delay in response.  Is this still an issue @ripples-alive? If so, can you tell me what you mean by an "information field"?

**Comment by ripples-alive** [comment]
@fletch3555 It's just a field called `information` in my code, no other means.
The key is that this field used the [attribute casting](https://laravel.com/docs/5.4/eloquent-mutators#attribute-casting) feature to be automatically converted to array.

**Comment by fletch3555** [comment]
Ah. Yeah, Voyager isn't going to know how to work with that. We can't possibly predict everything a dev will do

**Comment by ripples-alive** [comment]
Yeah, it's impossible to predict everything. But I don't think it's good to break down the whole page just because of one field is a good manner. If the field is not string, maybe you can try to serialize it with `json_encode`, or even allow user defined serializer(preprocessor).
You may assume that all the fields are `string`. If so, it's more reasonable to get value from the internal `$attributes` array instead of using getter.

**Comment by KgotsoK** [comment]
Also having the same issue. I think the "if array, covert to json" suggestion would work best.

**Comment by mabasic** [comment]
I am having issues with a database column of type JSON. I get the same error as mentioned in this issue.

Maybe add a column type for json and then display it in a textarea or code editor while converting from array to string  or something like that?

**Comment by GhoSTBG** [comment]
Any update? I have the same issue

**Comment by DimitarDechew** [comment]
Working solution is to use Laravel Mutator to get the information

In your Model:
```
public function getInformation()
{
    return json_decode($this->information, true);
}
```

**Comment by emptynick** [comment]
Well then you don't need to use casts at all, because the mutator will revert that.

**Comment by shaunthegeek** [comment]
I had fixed this issue as a patch `2018_11_26_144737_voyager_edit_json.patch`, I will create a pull request.

voyager patches: https://github.com/stonecutter/laravel-demo/tree/master/patches

**Comment by domainregistrar** [comment]
@sinkcup Was a PR ever made to merge this in? 

Would be great to see support for cast fields for json/array.

**Comment by shaunthegeek** [comment]
@domainregistrar Yes, there was a PR, but be closed during an argument.
you could use it quietly.

**Comment by almakano** [comment]
Any updates ?
